x-pack/legacy/plugins/maps/public/layers/styles/components/ranged_style_legend_row.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { EuiFlexGroup, EuiFlexItem, EuiText, EuiSpacer, EuiToolTip } from '@elastic/eui';
+
+export class RangedStyleLegendRow extends React.Component {
```

Comment:
Keep as functional component. These are more performant, https://reactjs.org/blog/2015/10/07/react-v0.14.html#stateless-functional-components

"This pattern is designed to encourage the creation of these simple components that should comprise large portions of your apps. In the future, we’ll also be able to make performance optimizations specific to these components by avoiding unnecessary checks and memory allocations.
"